File: java_integration/app/src/main/java/com/firewok/example/java_integration/feedintegration/fragment/FragmentIntegrationActivity.java
package com.firewok.example.java_integration.feedintegration.fragment;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.MenuItem;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import com.firewok.example.java_integration.R;
import com.firewok.example.java_integration.databinding.ActivityFragmentIntegrationBinding;

public class FragmentIntegrationActivity extends AppCompatActivity {
    private ActivityFragmentIntegrationBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityFragmentIntegrationBinding.inflate(LayoutInflater.from(this));
        setContentView(binding.getRoot());

        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setTitle(getString(R.string.fragment_screen_title));
            actionBar.setDisplayHomeAsUpEnabled(true);
            actionBar.setDisplayShowHomeEnabled(true);
        }

        if (savedInstanceState == null) {
            getSupportFragmentManager()
                    .beginTransaction()
                    .add(R.id.container, FeedViewFragment.newInstance(), null)
                    .commit();
        }
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            finish();
            return true;
        }
        return super.onOptionsItemSelected(item);
    }

    public static Intent getIntent(Context context) {
        return new Intent(context, FragmentIntegrationActivity.class);
    }
} 

File: feed_integration/app/src/main/java/tv/fw/feedintegration/fragment/FragmentIntegrationActivity.kt
package tv.fw.feedintegration.fragment

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import tv.fw.feedintegration.R
import tv.fw.feedintegration.databinding.ActivityFragmentIntegrationBinding

class FragmentIntegrationActivity : AppCompatActivity() {

    private lateinit var binding: ActivityFragmentIntegrationBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityFragmentIntegrationBinding.inflate(LayoutInflater.from(this))
        setContentView(binding.root)

        supportActionBar?.title = getString(R.string.fragment_screen_title)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar?.setDisplayShowHomeEnabled(true)
        if (savedInstanceState == null) {
            supportFragmentManager.beginTransaction()
                .add(R.id.container, FeedViewFragment.newInstance(), null)
                .commit()
        }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            finish()
        }
        return super.onOptionsItemSelected(item)
    }

    companion object {
        fun intent(context: Context) = Intent(context, FragmentIntegrationActivity::class.java)
    }
}
